client/src/main/scala/org/http4s/client/Client.scala

```diff
@@ -189,6 +190,15 @@ trait Client[F[_]] {
 
   @deprecated("Use expect", "0.14")
   def prepAs[T](req: F[Request[F]])(implicit d: EntityDecoder[F, T]): F[T]
+
+  /**
+    * Translates the effect type of this client from F to G
+    */
+  def imapK[G[_]: Defer: Bracket[?[_], Throwable]](fk: F ~> G)(gK: G ~> F)(
```

Comment:
The `Defer` and `Bracket` constraint is interesting.  On server, it's `Sync`.  Is there anything that should have both these constraints that would not be `Sync`?

Some prior discussion of `Defer` in http4s: https://github.com/http4s/http4s/pull/2325

Reply:
Not that I know of. It seems like similar considerations apply here as in the discussion you linked, so I've changed it for consistency.